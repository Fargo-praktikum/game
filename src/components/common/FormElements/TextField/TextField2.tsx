import React from 'react';

import "./TextField.scss";
import {FieldProps} from "formik";

type OwnProps = FieldProps<string, any>;
export const TextField2: React.FC<OwnProps> = ({field, form, meta, ...props}) => {
    const {name} = field;
    const hasError = form.touched[name] && form.errors[name];

    return (
        <div className={`formControl ${hasError ? 'errorInput' : ''}`}>
            <input className="textField"
                {...field} {...props} />
            <div className="errorContainer">
                {hasError && (
                    <span className="error">{form.errors[name]}</span>
                )}
            </div>

        </div>
    );
};


