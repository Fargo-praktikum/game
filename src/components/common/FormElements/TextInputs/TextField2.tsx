import React from "react";

import "./TextInputs.scss";
import { FieldProps } from "formik";

type OwnProps = FieldProps< string, any>;

export const TextField2 = ({ field, form, meta, ...props }: OwnProps) => {
    const { name } = field;
    const hasError = form.touched[name] && form.errors[name];

    return (
        <div className={`formControl ${hasError ? "errorInput" : ""}`}>
            <input className="textField"
                {...field} {...props} />
            <div className="errorContainer">
                {hasError && (
                    <span className="error">{form.errors[name]}</span>
                )}
            </div>

        </div>
    );
};


