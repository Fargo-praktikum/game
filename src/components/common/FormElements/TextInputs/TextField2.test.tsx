import React from "react";

import renderer from "react-test-renderer";
import { TextField2 } from "./TextField2";

describe("Проверка TextField2", () => {
    it("корректная отрисовка", () => {
        const tree = renderer.create(<TextField2/>).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("наличие всех свойств", () => {
        const field = { value: "testValue", name: "test", onChange: () => {return;}, onBlur: () => {return;} };
        const tree = renderer.create(<TextField2
            field={field}
        />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});



