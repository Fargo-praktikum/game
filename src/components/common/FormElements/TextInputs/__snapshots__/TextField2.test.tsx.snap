// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Проверка TextField2 корректная отрисовка 1`] = `
<div
  className="formControl "
>
  <input
    className="textField"
  />
  <div
    className="errorContainer"
  />
</div>
`;

exports[`Проверка TextField2 наличие всех свойств 1`] = `
<div
  className="formControl "
>
  <input
    className="textField"
    name="test"
    onBlur={[Function]}
    onChange={[Function]}
    value="testValue"
  />
  <div
    className="errorContainer"
  />
</div>
`;
